packages/core-bridge/src/lib.rs

```diff
@@ -243,6 +250,32 @@ fn start_bridge_loop(
                                     .await;
                                 break;
                             }
+                            Request::PollLogs { callback } => {
+                                let logs = core.fetch_buffered_logs();
+                                if !logs.is_empty() {
+                                    send_result(event_queue.clone(), callback, |cx| {
+                                        let logarr = cx.empty_array();
+                                        for (i, cl) in logs.into_iter().enumerate() {
+                                            let logobj = cx.empty_object();
+                                            let level = cx.string(cl.level.to_string());
+                                            logobj.set(cx, "level", level).unwrap();
+                                            // eww, time conversion. Use neon date if possible in napi-5
+                                            let ts = cx.number(cl.millis_since_epoch() as f64);
+                                            logobj.set(cx, "timestamp_nanos", ts).unwrap();
+                                            let msg = cx.string(cl.message);
+                                            logobj.set(cx, "message", msg).unwrap();
+                                            logarr.set(cx, i as u32, logobj).unwrap();
+                                        }
+                                        Ok(logarr)
+                                    });
+                                } else {
+                                    // Dispose of unused root
+                                    event_queue.send(move |mut cx| {
```

Comment:
I think callbacks should always be called otherwise you're left with unresolved promises in JS.

Reply:
K, I'll just call it with the empty array then